File: 01_validacao_de_nome/validar-username.dart
import 'dart:io';

bool usernameValidation(String str) {
  if (str.length >= 4 && str.length <= 25 && 
  RegExp(r'^[a-zA-Z]').hasMatch(str) &&
  RegExp(r'^[a-zA-Z0-9_]*$').hasMatch(str) &&
  !str.endsWith('_')) {
    return true
  } else {
    return false
  }
}

File: 01_easy/01_validacao_de_nome/validar-username.py
def UserNameValidation(str):
  isValidNUsername = 4 <= len(str) <= 25 and str[0].isalpha() and str.isalnum() and not str.endswith('_')
  if isValidNUsername:
    return True
  else:
    return False

name = 'leo_silva'
print(UserNameValidation(name))
